lib/collection/src/collection.rs

```diff
@@ -258,17 +258,79 @@ impl Collection {
         todo!("Handle for sender and receiver")
     }
 
+    pub async fn initiate_temporary_shard(&self, shard_id: ShardId) -> CollectionResult<()> {
+        let shard_holder_read = self.shards_holder.read().await;
+        // validate that the shard is known as a remote shard
+        if let Some(Shard::Remote(_)) = shard_holder_read.get_shard(&shard_id) {
+            // do not lock shards while creating temporary shard
+            drop(shard_holder_read);
+            let shard_path = Self::create_temporary_shard_dir(&self.path, shard_id).await?;
+            let temporary_shard =
+                LocalShard::build(shard_id, self.id.clone(), &shard_path, self.config.clone())
+                    .await?;
+            // short write lock to insert the temporary shard
+            let mut shards_holder_write = self.shards_holder.write().await;
+            shards_holder_write.add_temporary_shard(shard_id, temporary_shard);
+            Ok(())
+        } else {
+            Err(CollectionError::BadRequest {
+                description: "A shard transfer must target an existing remote shard".to_string(),
+            })
+        }
+    }
+
+    /// Create directory to hold the temporary data for shard with `shard_id`
+    pub async fn create_temporary_shard_dir(
+        collection_path: &Path,
+        shard_id: ShardId,
+    ) -> CollectionResult<PathBuf> {
+        let shard_path = collection_path.join(format!("{shard_id}-temp"));
+        tokio::fs::create_dir_all(&shard_path)
+            .await
+            .map_err(|err| CollectionError::ServiceError {
+                error: format!(
+                    "Can't create shard {shard_id} temporary directory. Error: {}",
```

Comment:
that might be no so strictly required

Reply:
https://github.com/qdrant/qdrant/pull/829/commits/684731b622a27cdda0e5a6725f066f0919945553